utils/zerotrie/src/cursor.rs

```diff
@@ -222,6 +222,97 @@ impl<'a> ZeroTrieSimpleAsciiCursor<'a> {
         reader::take_value(&mut self.trie.store)
     }
 
+    /// Probes the next byte in the cursor, stepping the cursor and returning the byte.
+    ///
+    /// This function is similar to [`Self::step()`], but it takes an index instead of a char.
+    /// This enables stepwise iteration over the contents of the trie.
```

Comment:
I don't think I understand what this means

Reply:
Can you suggest wording that explains it better? This was my attempt at explaining it clearly+succinctly

Reply:
I can't suggest wording because I don't understand what this means :)

I'm looking at step and still not understanding how these relate. I suspect wording it less dependently on step will help, and you can mention the correlation after that.

Reply:
I reworded the first paragraph of the doc string. Does that help explain the second paragraph?

Reply:
If there is only one child node, `step` and `probe` have the same behavior. If there are multiple children, `probe` can walk them by their index, whereas `step` binary searches to find the desired child.

Reply:
yeah this works